src/components/Avatar/Avatar.tsx

```diff
@@ -21,16 +21,24 @@ import { SuspenseImg } from "../../utils/SuspenseImg";
 import styles from "./Avatar.module.css";
 import { useIdColorHash } from "./useIdColorHash";
 
-type AvatarProps = JSX.IntrinsicElements["span"] & {
+type AvatarProps = (
+  | JSX.IntrinsicElements["button"]
+  | JSX.IntrinsicElements["span"]
+) & {
+  as: "button" | "span";
   src?: React.ComponentProps<typeof SuspenseImg>["src"];
   id: string;
   name: string;
   type?: "square" | "round";
   size?: CSSStyleDeclaration["height"];
+  onClick: (e: React.MouseEvent) => void;
```

Comment:
Can we please protect against span+onClick at the type level, this is poor for a11y

Reply:
That was my original intent, to use type unions, but the combination of everything i tried resulted with TS being very unhappy.

Reply:
If you cannot do it as such then maybe the combination is bad and you should be more declarative and need to wrap it in a semantic button component as the caller?

Reply:
I have removed the `as`, and infer the HTML tag from the `onClick` prop. 
If we're using `onClick` then the avatar switches to be a `button`